src/main/scala/io/qbeast/utils/QbeastUtils.scala

```diff
@@ -132,4 +150,51 @@ object QbeastUtils extends Logging {
     quantilesArray.mkString("[", ", ", "]")
   }
 
+  @Experimental
+  @nowarn("cat=deprecation")
+  def updateTransformationTypes(revision: Revision): Revision = {
+    val updatedTransformers = revision.columnTransformers.map {
+      case s: StringHistogramTransformer =>
+        CDFQuantilesTransformer(s.columnName, StringDataType)
+      case other => other
+    }
+    val updatedTransformations = revision.transformations
+      .zip(revision.columnTransformers)
+      .map {
+        case (s: StringHistogramTransformation, _) =>
+          CDFStringQuantilesTransformation(s.histogram)
+        case (i: IdentityToZeroTransformation, linearTransformer: LinearTransformer) =>
+          IdentityTransformation(
+            i.identityValue,
+            linearTransformer.dataType.asInstanceOf[OrderedDataType])
+        case (_ @NullToZeroTransformation, linearTransformer: LinearTransformer) =>
+          IdentityTransformation(null, linearTransformer.dataType.asInstanceOf[OrderedDataType])
+        case (otherTransformation, _) => otherTransformation
+      }
+    revision
+      .copy(columnTransformers = updatedTransformers, transformations = updatedTransformations)
+  }
+
+  @Experimental
+  def updateTransformationTypes(qbeastTable: QbeastTable): Unit = {
```

Comment:
We should ensure we don't update the metadata if there are no revision changes.

Reply:
done in [30b3e1e](https://github.com/Qbeast-io/qbeast-spark/pull/485/commits/30b3e1e315c051bd26cfb51ec7f9b05ff9e73ba7)